composables/useCards.ts

```diff
@@ -0,0 +1,82 @@
+import { useCardFieldTypes } from "~/composables/useCardFieldTypes";
+import type { Type } from "~/types/models/Card";
+import { useFieldTypeStore } from "~/store/fieldTypeStore";
+
+export function useCards() {
+  const cardStore = useCardStore();
+  const { fieldTypes } = storeToRefs(useFieldTypeStore());
+  const { fetchData } = useCardFieldTypes();
+
+  onMounted(async () => await fetchData());
+
+  const { label, personalFields, generalFields, socialFields, messagingFields, businessFields } = storeToRefs(cardStore);
+
+  const getFieldTypesId = (name: string) => fieldTypes.value.find((fieldType: Type) => fieldType.name === name)?.id ?? null;
+  const isNotEmpty = (value: any) => value !== null && value !== undefined && value !== "";
+  const getValuesByType = (fields: any[], name: string): string => {
+    return fields
+      .filter((field: any) => field.name === name)
+      .map((field: any) => field.value)
+      .join(', ');
+  };
+
+  const getLabelByType = (fields: any[], name: string): string => {
+    return fields
+      .filter((field: any) => field.name === name)
+      .map((field: any) => field.label)
+      .join(', ');
+  };
+
+  const cardData = computed(() => {
+    const personalField = ["Name", "Job Title", "Department", "Company Name"]
+    .map(name => ({ type_id: getFieldTypesId(name), value: getValuesByType(personalFields.value, name) }))
+    .filter(field => isNotEmpty(field.value));
+  
+    const generalField = ["Email", "Phone", "Company URL", "Link", "Address"]
+      .map(name => ({
+        type_id: getFieldTypesId(name),
+        value: getValuesByType(generalFields.value, name),
+        label: getLabelByType(generalFields.value, name)
+      }))
+      .filter(field => isNotEmpty(field.value));
+    
+    const socialField = ["Facebook", "X", "Instagram", "Threads", "LinkedIn", "Youtube", "Tiktok", "Spotify"]
+      .map(name => ({
+        type_id: getFieldTypesId(name),
+        value: getValuesByType(socialFields.value, name),
+        label: getLabelByType(socialFields.value, name)
+      }))
+      .filter(field => isNotEmpty(field.value));
+    
+    const messagingField = ["Skype", "Telegram", "Whatsapp"]
+      .map(name => ({
+        type_id: getFieldTypesId(name),
+        value: getValuesByType(messagingFields.value, name),
+        label: getLabelByType(messagingFields.value, name)
+      }))
+      .filter(field => isNotEmpty(field.value));
+    
+    const businessField = ["Calendly"]
+      .map(name => ({
+        type_id: getFieldTypesId(name),
+        value: getValuesByType(businessFields.value, name),
+        label: getLabelByType(businessFields.value, name)
+      }))
+      .filter(field => isNotEmpty(field.value));
+
+    const fields = [
+      ...personalField,
+      ...generalField,
+      ...socialField,
+      ...messagingField,
+      ...businessField
+    ];
+
+    return { label: label.value || "", fields };
+  });
+
+  return { cardData };
+}
```

Comment:
@KaelNierras 

Any reason why we have mapping of personalField, generalField, socialField, etc here 


on the list of card field types we have category already

![image](https://github.com/user-attachments/assets/c98269f7-974d-42f5-b249-be1b11f23fe8)

Reply:
The function is used to get the current data from card store and will be send to the BE

Reply:
Already revised please check